folding-schemes/src/frontend/noname/utils.rs

```diff
@@ -0,0 +1,157 @@
+use crate::Error;
+use ark_ff::PrimeField;
+use ark_r1cs_std::fields::fp::FpVar;
+use ark_r1cs_std::R1CSVar;
+use ark_relations::r1cs::SynthesisError;
+use noname::inputs::{parse_inputs, JsonInputs};
+use num_bigint::BigUint;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+pub const NONAME_CIRCUIT_EXTERNAL_INPUTS: &str =
+    "fn main(pub ivc_input: [Field; 2], external_inputs: [Field; 2]) -> [Field; 2] {
+    let xx = external_inputs[0] + ivc_input[0];
+    let yy = external_inputs[1] * ivc_input[1];
+    assert_eq(yy, xx);
+    return [xx, yy];
+}";
+
+pub const NONAME_CIRCUIT_NO_EXTERNAL_INPUTS: &str =
+    "fn main(pub ivc_input: [Field; 2]) -> [Field; 2] {
+    let out = ivc_input[0] * ivc_input[1];
+    return [out, ivc_input[1]];
+}";
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NoNameJSONIVCInput {
+    pub ivc_input: Vec<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NoNameJSONExternalInputs {
+    pub external_inputs: Vec<String>,
+}
+
+impl<F: PrimeField> From<Vec<F>> for NoNameJSONIVCInput {
+    fn from(value: Vec<F>) -> Self {
+        let mut ivc_input = Vec::with_capacity(value.len());
+        for field_element in value {
+            match field_element == F::zero() {
+                true => ivc_input.push("0".to_string()),
+                false => ivc_input.push(field_element.to_string()),
+            }
+        }
+        NoNameJSONIVCInput { ivc_input }
+    }
+}
+
+impl NoNameJSONIVCInput {
+    pub fn from_fpvars<F: PrimeField>(value: &Vec<FpVar<F>>) -> Result<Self, SynthesisError> {
+        let mut field_elements = Vec::<F>::with_capacity(value.len());
+        for var in value {
+            let val = var.value()?;
+            field_elements.push(val);
+        }
```

Comment:
I suggest to go for iterators rather than a for loop when the loops aren't crazy.

This makes it easier to add/remove and understand what happens there.

Reply:
Yep, thanks for the suggestions, did this in the latest changes.